### PR TITLE
Stop creation of a pickled result file on client side

### LINKED ISSUE #256: Don't create local results file
Today, when you do `ct.dispatch()`, there is a pickle file created on the client side and then uploaded to the queuer. But when the dispatcher updates the results, that happens in the results service and the client side results file is not updated, thus causing a divergence problem.

The solution is to not create the results pickle file on the client side and instead send bytes to the queuer. 

There should also be a `ct.get_results()` function that returns the result file as bytes.

Related to #242

### PR DESCRIPTION
**Pull request by kessler-frost**
- [ ] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

Fixes #256.

### CODE REVIEW

File: refactor/client_side/interface_with_covalent.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+from functools import wraps
+from io import BytesIO
+
+import cloudpickle as pickle
+import requests
+
+from covalent._results_manager.result import Result
+from covalent._workflow.lattice import Lattice
+
+SUBMIT_DISPATCH_QUEUER_ADDR = "http://localhost:8000/api/v0/submit/dispatch"
```
Comment (FyzHsn) [inline refactor/client_side/interface_with_covalent.py:36]: Use the Settings class and `get_route` method. No more hardcoded file paths.

File: refactor/client_side/interface_with_covalent.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+GET_RESULT_ADDR = "http://localhost:8000//api/v0/workflow/results"
```
Comment (FyzHsn) [inline refactor/client_side/interface_with_covalent.py:37]: Same as above.

**Review by FyzHsn — CHANGES_REQUESTED** [review]

File: refactor/client_side/interface_with_covalent.py (hunk 1 above)
Comment (kessler-frost) [reply refactor/client_side/interface_with_covalent.py]: Updated.

File: refactor/client_side/interface_with_covalent.py (hunk 2 above)
Comment (kessler-frost) [reply refactor/client_side/interface_with_covalent.py]: Updated.

**Pull request merged**